src/Illuminate/Queue/Queue.php

```diff
@@ -209,4 +224,12 @@ public function setContainer(Container $container)
     {
         $this->container = $container;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setMetadata(Collection $metadata)
```

Comment:
Do we really need to force metadata to be a collection? Why cant it just be left as a generic variable, or an array?

Then I can use it as `$queue->setMetadata(['thing' => true])`

Reply:
The idea is you wouldn't directly call `setMetadata` and use the `metadata`  method on the queue manager which builds the collection. Doing it this way would allow third party packages to tap into the feature and not run the risk of it being overridden.

The only reason the `setMetadata` method exists is because I could not get to the queue manager from the Queue itself. So the metadata had to be passed down on creation.

Happy to flesh this out though.

Reply:
It's just that later on, when you push the data to the queue Payload - you call `toArray()` - so in essence we seem to be converting to a Collection - then converting back to an Array? So is there much benefit in doing it vs staying as an array.

Reply:
Ok - yep - I understand what angle you are coming from now.